src/App.js

```diff
@@ -1,7 +1,25 @@
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import HomePage from "./pages/Home";
+import RootLayout from "./pages/Root";
+import AuthenticationPage from "./pages/Authentication";
+import ErrorPage from "./pages/Error";
+
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <RootLayout />,
+    errorElement: <ErrorPage />,
+    children: [
+      { index: true, element: <HomePage /> },
```

Comment:
Shouldn't we have an auth check here?

Reply:
The auth checking is done in the feature_subscription branch. At this moment, it simply just hit login and register endpoints. All the token storing and other authentication work is done in the subscription branch as we wanted to protect the subscription page from unauthorized access